lib/advanced_computer_receiver.rb

```diff
@@ -0,0 +1,111 @@
+class AdvancedComputerReceiver
+  def initialize(board, game_checker)
+    @board = board
+    @game_checker = game_checker
+  end
+
+  def get_player_input(validator, display, computer_player, human_player)
+    calculate_best_move(validator, computer_player, human_player)
+  end
+
+  private
+
+  def calculate_best_move(validator, computer_player, human_player)
+    # need to determine best move.
+    # computer win
+    # stop human win
+    # fork
+    # block fork
+    # center
+    # opposite corner
+    # empty corner
+    # empty side
+    corners = [1, 3, 7, 9]
+    sides = [2, 4, 6, 8]
+
+    if check_function(validator, computer_player, :winner?)
+      #   puts "computer win"
+      check_function(validator, computer_player, :winner?)
+    elsif check_function(validator, human_player, :winner?)
+      #   puts "block human"
+      check_function(validator, human_player, :winner?)
+    elsif check_function(validator, computer_player, :is_fork?)
+      #   puts "create fork"
+      check_function(validator, computer_player, :is_fork?)
+
+    elsif check_function(validator, human_player, :is_fork?)
+      #   puts "block fork"
+      check_function(validator, human_player, :is_fork?)
```

Comment:
Wow lots of if statements.  Could you use recursion here instead of calling check_function multiple times?

Reply:
....maybe? I will have to think about this.